.github/workflows/cd.yml

```
@@ -1,18 +1,79 @@
 name: Continuous Deployment
 
 on:
+  workflow_dispatch:
+  schedule:
+    # At 00:00 on Saturday
+    # https://crontab.guru/#0_0_*_*_6
+    - cron: "0 0 * * 6"
   push:
     tags:
       - "v*.*.*"
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
-  publish:
-    name: Publish on crates.io
+  publish-nightly:
+    name: Create a nightly release
     runs-on: ubuntu-latest
+    if: ${{ !startsWith(github.event.ref, 'refs/tags/v') }}
     steps:
       - name: Checkout the repository
         uses: actions/checkout@v3
-      - name: Publish
+        with:
+          fetch-depth: 0
+
+      - name: Calculate the next release
+        run: |
+          suffix="-alpha"
```

(nit: to go along with the suggestions below), it might be easier to drop the hyphen here and add it to the bits below
```suggestion
          suffix="alpha"
```